### DISCUSSION TITLE
Help with very high cpu usage

### DISCUSSION THREAD

**Discussion by cornzzy** [erebe/wstunnel#340] [Q&A]
I have the following setup:

`wireguard client` -> `server1` -> `server2`

`wiregaurd client` is simply pointing to server1, `server1` is running wstunnel client and `server2` is running wireguard and wstunnel server.

CPU usage hits 25% in server1 with a 1 core ($5/m server) when only 1 client using only 2 megabytes bandwidth. Is there a way to use wstunnel at scale?

I tried with ws no encryption and turned off logging and still really high cpu usage.

**Comment by erebe** [comment]
Like I don't really know. 
You can try to use linux perf, to see where all this CPU cycles are going into.
https://phoenixnap.com/kb/linux-perf#ftoc-heading-12

Be aware that cheap VPS kind of really cheat/squeeze the notion of what a vCPU is.  Aka, theuy are really low performance.
